client_app/src/components/shelter/ProcessVolunteerList.js

```diff
@@ -0,0 +1,76 @@
+import { Link } from "react-router-dom";
+
+const handleReload = () => {
+  window.location.reload();
+};
+
+const ProcessVolunteerList = ({shiftDetails}) => {
+    const currentTime = Date.now();
+    const filteredShifts = shiftDetails.filter(item => item.start_time < currentTime);
+    const workerList = filteredShifts.flatMap(item => {
+        const workers = item.worker ? item.worker.split(",").map(name => name.trim()) : [];
+        const emails = item.email ? item.email.split(",").map(email => email.trim()) : [];
+        
+        return workers.map((name, index) => ({
+          worker: name,
+          email: emails[index]
+        }));
+    });
+
+    const uniqueWorkerList = workerList.filter((entry, index, self) =>
+        index === self.findIndex(e => e.worker === entry.worker && e.email === entry.email)
+    );
+
+    return (
+      <>
+        <div className="conf-page">
+          <h1>Past Volunteers</h1>
+          <table>
+            <thead>
+              <tr>
+                <th>
+                  <h3>#</h3>
+                </th>
+                <th>
+                  <h3>Volunteer Name</h3>
+                </th>
+                <th>
+                  <h3>Volunteer Email</h3>
+                </th>
+              </tr>
+            </thead>
+            <tbody>
+              {uniqueWorkerList.length > 0 ? (uniqueWorkerList.map((list, index) => (
```

Comment:
Make this a constant and render like:
{uniqueList && //your components} 

This is more readable than using ? since the lines are so far apart.

Reply:
Since the table code is so long you could also make it its own method 

const renderTable = () => {
 return (//your code)
}
and then call renderTable()  in the block